File: examples/js/structs/src/parking-ticket-report.ts
import { LHStruct, LHField } from 'littlehorse-client'
import { VariableType } from 'littlehorse-client/proto'

@LHStruct({ name: 'car' })
export class ParkingTicketReport {
  @LHField(VariableType.STR)
  vehicleMake!: string

  @LHField(VariableType.STR)
  vehicleModel!: string

  @LHField(VariableType.STR)
  licensePlateNumber!: string

  @LHField(VariableType.STR)
  createdAt!: string

  constructor(vehicleMake?: string, vehicleModel?: string, licensePlateNumber?: string, createdAt?: string) {
    if (vehicleMake !== undefined) this.vehicleMake = vehicleMake
    if (vehicleModel !== undefined) this.vehicleModel = vehicleModel
    if (licensePlateNumber !== undefined) this.licensePlateNumber = licensePlateNumber
    if (createdAt !== undefined) this.createdAt = createdAt
  }

  toString(): string {
    return `${this.vehicleMake} ${this.vehicleModel}, Plate Number: ${this.licensePlateNumber}, issued at ${this.createdAt}`
  }
}

File: examples/go/structdef/structs/parking_ticket_report.go
package structs

import (
	"fmt"

	"github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"
)

const ParkingTicketReportStructDefName string = "parking-ticket-report"
const ParkingTicketReportStructDefDescription string = "ParkingTicketReport represents a parking ticket report for a vehicle."

// ParkingTicketReport represents a parking ticket report for a vehicle.
type ParkingTicketReport struct {
	VehicleMake        string `json:"vehicleMake"`
	VehicleModel       string `json:"vehicleModel"`
	LicensePlateNumber string `json:"licensePlateNumber"`
}

func (ParkingTicketReport) LHStructDef() littlehorse.LHStructDefInfo {
	return littlehorse.LHStructDefInfo{Name: ParkingTicketReportStructDefName, Description: ParkingTicketReportStructDefDescription}
}

func (r ParkingTicketReport) String() string {
	return fmt.Sprintf("%s %s, Plate Number: %s", r.VehicleMake, r.VehicleModel, r.LicensePlateNumber)
}
